alice is entering her new svg icon era

jennej opened the pull request.

jennej committed:
delete icons that aren't used anywhere

jennej committed:
new handle style icons

jennej committed:
new svg icons for gallery resources

-1 icon factory!
pretty crisp icons at any size!
begin consolidating the location of icons

jennej committed:
clean up icons/recycle bin

flatten directories, we're about to make everything svg anyway
remove unused icons
put the recycle bin icons that we actually use in icons

jennej committed:
corral the icons into a single place, use svg where i can

jennej committed:
new svg icons/hover behavior for link buttons

jennej committed:
svg undo/redo icons

jennej committed:
svg icons for personbuilder

jennej committed:
svg icons for object markers

jennej committed:
svg icons for camera/vruser marker icons

jennej committed:
svg icons for camera options

jennej committed:
svg icons for debug user history

jennej committed:
move "back" icon

jennej committed:
svg icons for camera controls

jennej committed:
svg icons for vr headset/hands

jennej committed:
svg folder icon

jennej committed:
consolidate gallery theme icon images with other icons

jennej committed:
consolidating icons in alice.org.ide

dabeshouse commented on core/ide/src/main/resources/org/alice/ide/icons/images/handles/slideUpRight.svg:
Rename slideUpRight. Currently throws an NPE.

dabeshouse commented on core/ide/src/main/java/org/alice/stageide/sceneeditor/interact/handles/ManipulationHandle2DCameraDriver.java:
```diff
@@ -42,25 +42,35 @@
+    BackRight("images/handles/driveBackRight.svg"),
+    Forward("images/handles/driveForward.svg"),
+    ForwardLeft("images/handles/driveForwardLeft.svg"),
+    ForwardRight("images/handles/driveForwardRight.scg"),
```
Should be svg.

jennej replied:
oops, thx

dabeshouse commented on core/ide/src/main/resources/org/alice/ide/icons/images/markers/VrMarkerIconGrayScale.svg:
We don't seem to use this. Do we want to keep it?

jennej replied:
oh, no, i should have deleted this

dabeshouse commented on core/ide/src/main/resources/org/alice/ide/icons/images/completed.svg:
I think this is only used in the developer tool UserActivity History.
In that I see a red line above or below the check mark. Often on many of them. Expanding, collapsing, or other interactions usually make the red go away..
This is not a problem, but if you have any idea why I am curious.

jennej replied:
oh weird. i wasn't seeing it all until i scrolled.  it appears to only show up on things we've scrolled out of view and then back in?!

dabeshouse replied:
I guess I scrolled to cause it as well. It is odd, but not seen by anyone but us, and not very often there.

dabeshouse reviewed (changes requested):
So much nicer.
Two questions to consider, two name mismatches to fix, then good to go.

jennej committed:
pr feedback/ typo fixes

jennej committed:
don't turn svgs to images in the moveToImageIcon

also:
fixed the spacing issue the correct way instead of the hack I had
centered the arrow icon vertically

dabeshouse reviewed (approved):
lgtm
